__tests__/middleware-test.js

```diff
@@ -0,0 +1,158 @@
+    return {
+        boardSlottingMarshaller: jest.fn(),
+        networkConfig: { rpcAddrs: ["http://127.0.0.1:26657"], chainName: "agoriclocal" },
+        makeAgoricNames: jest.fn()
+    };
+});
+
+jest.mock('../src/lib/wallet.js', () => {
+    return {
+        getCurrent: jest.fn(),
+    };
+});
+
+jest.mock('../src/oracle/helper.js', () => {
+    const originalModule = jest.requireActual("../src/oracle/helper")
+    return {
+        ...originalModule,
+        readVStorage: jest.fn()
+    };
+});
+
+import { readJSONFile, readVStorage } from '../src/oracle/helper.js';
+import { getOffers, checkSubmissionForRound, queryPrice, queryRound, getOraclesInvitations } from '../src/oracle/middleware';
+import { iterateReverse, makeLeader, makeFollower } from '@agoric/casting';
+import { boardSlottingMarshaller, makeAgoricNames } from '../src/lib/rpc.js'
+import { getCurrent } from '../src/lib/wallet.js'
+
+let iterateReverseOutput = readJSONFile("__tests__/mock-objects/followerReversed.json")
+let currentOutput = readJSONFile("__tests__/mock-objects/current.json")
+let agoricNamesOutput = readJSONFile("__tests__/mock-objects/agoricNames.json")
+
+const asyncIterator = {
+    [Symbol.asyncIterator]: function() {
+    let i = 0;
+    return {
+        next: () => {
+        return new Promise(resolve => {
+            if (i < iterateReverseOutput.length) {
+            resolve({ value: iterateReverseOutput[i++], done: false });
+            } else {
+            resolve({ done: true });
+            }
+        });
+        }
+    };
+    }
+};
+
+iterateReverse.mockReturnValue(asyncIterator)
+makeLeader.mockReturnValue({})
+makeFollower.mockReturnValue({})
+boardSlottingMarshaller.mockReturnValue({})
+getCurrent.mockReturnValue(currentOutput)
+makeAgoricNames.mockReturnValue(agoricNamesOutput)
+
+/**
+ * Test for get offers
+ */
+test('calls getOffers to get latest successful submissions', async () => {
+  
+    const offers = await getOffers({});
+  
+    //notice how only the first offer should be calculated because for every offer there will be 3 entries
+    expect(offers).toStrictEqual(
+    [
+        {
+            "status": {
+                "id": 1676450914983,
+                "invitationSpec": {
+                    "invitationArgs": [
+                        {
+                            "roundId": 735,
+                            "unitPrice": "13312851"
+                        }
+                    ],
+                    "invitationMakerName": "PushPrice",
+                    "previousOffer": 1675687989744,
+                    "source": "continuing"
+                },
+                "numWantsSatisfied": 1,
+                "payouts": {},
+                "proposal": {}
+            },
+            "updated": "offerStatus"
+        }
+    ]
+    );
+});
+
+/**
+ * Test for check submission - with submission
+ */
+test('calls checkSubmissionForRound to check if there was a submission for round with submission', async () => {
```

Comment:
doesn't fit 80 char limit

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/e7d6c899fec472591f9ed9fa8dfad2c660fbb9a5